cpp/leaky_core.h

```diff
@@ -54,35 +56,29 @@ class TermState {
   }
 };
 
-class SignalState {
+class SignalHandler {
  public:
-  SignalState()
-      : traps(NewDict<int, builtin_trap::_TrapHandler*>()),
-        nodes_to_run(NewList<syntax_asdl::command_t*>()) {
-  }
-  void InitShell() {
-  }
-  int GetLastSignal() {
-    return last_sig_num;
-  }
-  void AddUserTrap(int sig_num, builtin_trap::_TrapHandler* handler) {
-    NotImplemented();
-  }
-  void RemoveUserTrap(int sig_num) {
-    NotImplemented();
-  }
-  List<syntax_asdl::command_t*>* TakeRunList() {
-    return NewList<syntax_asdl::command_t*>();
-  }
-  int last_sig_num = 0;
-  Dict<int, builtin_trap::_TrapHandler*>* traps;
-  List<syntax_asdl::command_t*>* nodes_to_run;
+  SignalHandler();
+  void EnqueueSignal(int sig_num);
+  List<int>* TakeSignalQueue();
 
-  DISALLOW_COPY_AND_ASSIGN(SignalState)
+  List<int>* signal_queue;
```

Comment:
Please add a trailing underscore for members -- we're using the style consistently

(It basically comes from https://google.github.io/styleguide/cppguide.html , which is pretty  widely used )

Reply:
Ah right. Good catch